astroid/inference.py

```diff
@@ -260,16 +262,22 @@ def infer_call(self, context=None):
                 yield from callee.infer_call_result(caller=self, context=callcontext)
         except InferenceError:
             continue
-    return dict(node=self, context=context)
 
 
 nodes.Call._infer = infer_call  # type: ignore[assignment]
 
 
 @decorators.raise_if_nothing_inferred
 @decorators.path_wrapper
-def infer_import(self, context=None, asname=True):
+def infer_import(
+    self: nodes.Import,
+    context: InferenceContext | None = None,
+    asname: bool = True,
+    **kwargs: Any,
+) -> Generator[nodes.Module, None, None]:
     """infer an Import node: return the imported module/object"""
+    if not context:
```

Comment:
This will be uncovered but I think it makes more sense to handle the `context == None` path than see why `infer_import` and `infer_importfrom` are never called without a `context`.